clang/include/clang/Basic/Attr.td

```diff
@@ -3327,6 +3330,15 @@ def DiagnoseIf : InheritableAttr {
   let Documentation = [DiagnoseIfDocs];
 }
 
+def DiagnoseSpecializations : InheritableAttr {
+  let Spellings = [Clang<"diagnose_specializations", /*AllowInC*/0>];
```

Comment:
Should this be named `diagnose_explicit_specializations` given: https://eel.is/c++draft/temp.spec#general-4.sentence-2

Reply:
Do explicit specializations include partial specializations?